app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEIViewModel.kt

```diff
@@ -275,7 +275,11 @@ class SearchTEIViewModel(
         val updatedItems = uiState.items.map {
             (it as FieldUiModelImpl).copy(value = null, displayName = null)
         }
-        uiState = uiState.copy(items = updatedItems)
+        uiState = uiState.copy(
+            items = updatedItems,
+            searchedItems = mapOf(),
+        )
+        searching = false
```

Comment:
could you cover this change with a test, maybe performing the clear and then asserting the uiState?